spec/components/hint_spec.rb

```diff
@@ -8,6 +8,24 @@ def component_name
   it "renders hint" do
     render_component(text: "For example, ‘QQ 12 34 56 C’.")
 
-    assert_select(".govuk-hint", text: "For example, ‘QQ 12 34 56 C’.")
+    assert_select '.govuk-hint.govuk-\!-margin-bottom-3', text: "For example, ‘QQ 12 34 56 C’."
+  end
+
+  it "applies a specified bottom margin" do
+    render_component(text: "For example, ‘QQ 12 34 56 C’.", margin_bottom: 7)
+
+    assert_select '.govuk-hint.govuk-\!-margin-bottom-7'
+  end
+
+  it "applies zero bottom margin" do
+    render_component(text: "For example, ‘QQ 12 34 56 C’.", margin_bottom: 0)
```

Comment:
I was wondering if the API could avoid CSS and be 'spacing_bottom' but this feels as explicit as possible so 👍

Reply:
I don't know of an example of it, but at some point there might be a component where we might want to control `padding_bottom` for some reason, being this explicit seemed like a good approach.